ParatrooperUnity/Assets/Scripts/Gun.cs

```diff
@@ -38,8 +38,8 @@ public void Fire(Transform gunJoint)
 
 		var direction = (Vector2)(bulletSpawn.position - gunJoint.position).normalized;
 		var spawnPos = bulletSpawn.position;
-		var bullet = Instantiate(bulletPrefab, spawnPos, Quaternion.identity);
-		bullet.AddForce(direction * shotVelocity);
+		var bullet = PoolingFactory.SpawnOrRecycle(bulletPrefab.transform, spawnPos);

```

Comment:
Add a generic method to for SpawnOrRecycle that can instantiate it and return the correct type by using the Instantiate<> method. That way the factory will return the object as the correct type.